### ISSUE TITLE
suspicious error log: subchannel_list.h:535:31: runtime error: downcast of address

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#16632]
We keep seeing this on foundry test `//test/core/end2end:h2_http_proxy_test@bad_ping@poller=poll-cv` (and possibly on other test cases)

https://source.cloud.google.com/results/invocations/e858b6e2-214a-49f7-837f-90cbea0d15b7/targets/%2F%2Ftest%2Fcore%2Fend2end:h2_http_proxy_test@bad_ping@poller%3Dpoll-cv/log

```
src/core/ext/filters/client_channel/lb_policy/subchannel_list.h:535:31: runtime error: downcast of address 0x7f5030001d40 which does not point to an object of type 'grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelList'
0x7f5030001d40: note: object is of type 'grpc_core::SubchannelList<grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelList, grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelData>'
 50 7f 00 00  38 d3 33 01 00 00 00 00  28 bc 0d 02 00 00 00 00  01 00 00 00 00 00 00 00  f0 18 00 30
              ^~~~~~~~~~~~~~~~~~~~~~~
              vptr for 'grpc_core::SubchannelList<grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelList, grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelData>'!--
```

**Comment by dgquintas** [comment]
This looks very similar to #15410

**Comment by markdroth** [comment]
Yes, it's the same issue.  It was fixed in #16599.
